### PR TITLE
[Prim] Prim skip dynamic shape op by setting flag

### PR DESCRIPTION
**Pull request by cyber-pioneer**
### PR types
New features

### PR changes
Others

### Description
Pcard-66975
In default, prim would raise error when encountering dynamic shape.
By setting flag **FLAGS_prim_skip_dynamic=1**, prim will skip op with dynamic shape and raise warning.

### CODE REVIEW

**Review by Charles-hit — DISMISSED** [review]

File: python/paddle/autograd/backward_utils.py [hunk 1]
```diff
@@ -68,3 +72,33 @@ def copy(self, new_block):
...
+                warnings.warn(
+                    f"[Prim] Decomp op does not support dynamic shape -1, but got shape {item.shape} in inputs of op {op.name()} . Prim will skip its vjp op."
+                )
+                return True
+
+
+# Prim currently does not support dynamic shape, when dynamic shape exits in shape of op inputs, prim will be skipped its vjp op.
+@signature_safe_contextmanager
+def vjp_guard(op, inputs):
+    skip_prim_dynamic = (
```
Comment (Charles-hit) [inline python/paddle/autograd/backward_utils.py]:
```suggestion
    skip_prim = (
```

File: python/paddle/base/core.py [hunk 2]
```diff
@@ -513,6 +513,14 @@ def _get_batch_norm_none_var(op):
 }
 
 
+# This api is used for development for dynamic shape in prim, and will be removed in future.
+def _enable_prim_dynamic_shape():
+    if os.getenv("FLAGS_prim_dynamic") == "1":
```
Comment (Charles-hit) [inline python/paddle/base/core.py]:
```suggestion
    if os.getenv("FLAGS_enable_prim_dynamic") == "1":
```

Comment (cyber-pioneer) [reply python/paddle/base/core.py]: FLAGS_prim_dynamic -> FLAGS_prim_skip_dynamic

File: python/paddle/autograd/backward_utils.py (hunk 1 above)
Comment (cyber-pioneer) [reply python/paddle/autograd/backward_utils.py]: done

File: python/paddle/autograd/ir_backward.py [hunk 3]
```diff
@@ -678,13 +680,14 @@ def append_yield(block, inputs):
...
                         # create grad_op
                         before_ops_num = len(bwd_block.ops)
-                        input_grads = paddle.framework.core.call_vjp(
-                            op,
-                            inputs,
-                            outputs,
-                            output_grads,
-                            input_grad_stopgradients,
-                        )
+                        with dynamic_shape_prim_vjp_guard(op, inputs):
```
Comment (xiaoguoguo626807) [inline python/paddle/autograd/ir_backward.py:683]: 这个是当op有-1shape 时关闭组合选择大kernel吗

**Review by xiaoguoguo626807 — APPROVED** [review]

File: python/paddle/autograd/ir_backward.py (hunk 3 above)
Comment (cyber-pioneer) [reply python/paddle/autograd/ir_backward.py:683]: 是的，组合机制目前不支持-1的shape。

**Pull request merged**